### PR TITLE
Improvements required for new PubSub tests

### PR DESCRIPTION
**Pull request by fenek**
This PR introduces:

* A hack that fixes a rendering of stanza log link in failed cases (for OTP >= 18.3).
* MFA overridable type.
* New PubSub stanzas.
* Exported function from `escalus_story` that makes friends with `#client{}` handles instead of user atoms.

### CODE REVIEW

File: src/escalus_ct.erl [hunk 1]
```diff
@@ -101,3 +105,42 @@ do_log_stanza(Jid, Direction, Stanza) ->
...
+
+ct_int_header() ->
+    "</pre><div class=\"ct_internal\"><b>*** CT ~s *** ~ts</b>".
+
+ct_int_footer() ->
+    "</div><pre>".
+
+ct_log_timestamp({MS, S, US}) ->
+    put(log_timestamp, {MS, S, US}),
+    {{Year, Month, Day}, {Hour, Min, Sec}} = calendar:now_to_local_time({MS,S,US}),
```
Comment (gadgetci) [inline src/escalus_ct.erl]: According to **Elvis**:
> Missing space right "," on line 141

Comment (gadgetci) [inline src/escalus_ct.erl]: According to **Elvis**:
> Missing space right "," on line 141

File: src/escalus_story.erl [hunk 2]
```diff
@@ -57,6 +58,16 @@ make_everyone_friends(Config0, Users) ->
...
+    make_all_clients_friends(Clients),
+
+    % stop the clients
+    escalus_cleaner:clean(Config1),
+    escalus_cleaner:stop(Config1),
+
+    % return Config0
+    [{everyone_is_friends, true} | Config0].
+
+make_all_clients_friends(Clients) ->
```
Comment (gadgetci) [inline src/escalus_story.erl]: According to **Compiler**:
> Warning: missing specification for function make_all_clients_friends/1

**Comment by michalwski** [comment]
Thanks!

**Comment by arcusfelis** [comment]
@fenek 

```erlang
1> erlang:system_info(version).
"7.3.1.2"
2>     VerFloat = (catch list_to_float(erlang:system_info(version))),                   
2>     is_float(VerFloat) andalso VerFloat >= 7.3. 
false
3> 
3> VerFloat.
{'EXIT',{badarg,[{erlang,list_to_float,["7.3.1.2"],[]},
                 {erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,674}]},
                 {erl_eval,expr,5,[{file,"erl_eval.erl"},{line,431}]},
                 {erl_eval,expr,5,[{file,"erl_eval.erl"},{line,438}]},
                 {shell,exprs,7,[{file,"shell.erl"},{line,686}]},
                 {shell,eval_exprs,7,[{file,"shell.erl"},{line,641}]},
                 {shell,eval_loop,3,[{file,"shell.erl"},{line,626}]}]}}
```

**Comment by arcusfelis** [comment]
This should work
```erlang
lists:map(fun erlang:list_to_integer/1, string:tokens(erlang:system_info(version), ".")) >= [7,3].
```

**Comment by fenek** [comment]
@arcusfelis 
Indeed, I haven't considered such versions. ;< I'll patch it immediately.

**Pull request merged**